test/test_torch.py

```diff
@@ -1043,6 +1043,56 @@ def test_pairwise_distance_empty(self):
             self.assertEqual(torch.zeros(0, device=device), torch.pairwise_distance(x, y))
             self.assertEqual(torch.zeros((0, 1), device=device), torch.pairwise_distance(x, y, keepdim=True))
 
+    @skipIfRocm
+    def test_pdist_empty(self):
+        devices = ['cpu']
+        for device in devices:
+            shape = (0, 2)
+            x = torch.randn(shape, device=device)
+            self.assertEqual(torch.empty(0, device=device), torch.pdist(x))
+
+            shape = (1, 2)
+            x = torch.randn(shape, device=device)
+            self.assertEqual(torch.empty(0, device=device), torch.pdist(x))
+
+            shape = (3, 0)
+            x = torch.randn(shape, device=device)
+            self.assertEqual(torch.zeros(3, device=device), torch.pdist(x))
+
+    @skipIfRocm
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_pdist_special(self):
+        devices = ['cpu']
+        for device in devices:
+            for shape in [(4, 5), (3, 2), (2, 1)]:
+                x = torch.randn(shape, device=device)
+                actual = torch.pdist(x, p=np.inf)
+                expected = torch.max(torch.abs(x[:, None] - x[None]), 2)[0][np.triu_indices(shape[0], k=1)]
+                self.assertEqual(expected.shape, actual.shape)
+                self.assertTrue(torch.allclose(expected, actual))
+
+                x = (torch.rand(shape, device=device) < .5).type(torch.float)
+                actual = torch.pdist(x, p=0)
+                expected = torch.sum(x[:, None] != x[None], 2)[np.triu_indices(shape[0], k=1)]
+                self.assertEqual(expected, actual)
+
+    @skipIfRocm
+    @unittest.skipIf(not TEST_SCIPY, "Scipy not found")
+    def test_pdist_scipy(self):
+        from scipy.spatial.distance import pdist
+        devices = ['cpu']
+        for device in devices:
+            for shape in [(4, 5), (3, 2), (2, 1)]:
+                for p in [1, 2, 3, 1.5, 2.5]:
```

Comment:
these don't seem to match up when p is inf -- is that a bug in scipy?

Reply:
The doc for pdist just says >= 1 and passes a float. The backend `minkowski` function specifically says an integer, but clearly works in (0, inf). For 0 it returns inf, for inf it returns 1.

Reply:
That's pretty strange on their side.  So, what do you think the correct thing to do here is?  Add your own check for 0 and inf that doesn't go through scipy (assuming that your implementation gives different results)?

Reply:
Yeah, that's why there's `test_pdist_special` which tests the "special" values of 0 and infinity.

Reply:
Ah, okay I didn't know what "special" meant.  Maybe a comment (or moving it after the normal tests) would help clarify.